flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificInputTypeStrategies.java

```diff
@@ -125,6 +125,19 @@ public static InputTypeStrategy windowTimeIndicator() {
      */
     public static final ArgumentTypeStrategy INDEX = new IndexArgumentTypeStrategy();
 
+    /** An {@link ArgumentTypeStrategy} that expects a percentage value between [0.0, 1.0]. */
+    public static ArgumentTypeStrategy percentage(boolean expectedNullability) {
+        return new PercentageArgumentTypeStrategy(expectedNullability);
+    }
+
+    /**
+     * An {@link ArgumentTypeStrategy} that expects an array of percentages with each element
+     * between [0.0, 1.0].
+     */
+    public static ArgumentTypeStrategy percentageArray(boolean expectedNullability) {
```

Comment:
Is it possible to replace this new strategy with a `and` strategy that combines `ArrayTypeStrategy` and `percentage`?

Reply:
I think `and` strategy describes a relationship like `a + b`, but here we need a inclusion relationship which has not been supported yet. 
For example, if we want a strategy like `ARRAY<NUMERIC>`, then we have to create a new strategy class specially or in a very complex way, like `or(explicit(ARRAY(DOUBLE())), explicit(ARRAY(FLOAT()))...)`. This can be an improvement in the future.
Back to our implementation, we have to perform literal value checking additionally, so I took the compromised and easy way which is creating a new strategy specially.

Reply:
Reasonable choice : )